### PR TITLE
Added 1.0.0-beta4

### PR DESCRIPTION
**Pull request by vlesierse**
This PR contains a container for beta 4 of ASP.NET 5. Unfortunately the aspnet/Home repo has been changed and gives problems with the CI scripts.

### CODE REVIEW

File: 1.0.0-beta4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    && git clone https://github.com/mono/mono.git \
+    && cd mono \
+    && git reset --hard 53dc56ee39a8e3b013231957aca4671b202c6410 \
+    && ./autogen.sh --prefix="/usr/local" \
+    && make \
+    && make install \
+    && cd .. \
+    && rm mono -r
+
+# Install aspnet 1.0.0-beta4
```
Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: comment obsolete, actually doesn't do any installation in this block

File: 1.0.0-beta4/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get update -qq \
+    && apt-get install -qqy git autoconf libtool automake build-essential mono-devel gettext unzip \
+    && git clone https://github.com/mono/mono.git \
+    && cd mono \
+    && git reset --hard 53dc56ee39a8e3b013231957aca4671b202c6410 \
+    && ./autogen.sh --prefix="/usr/local" \
+    && make \
+    && make install \
+    && cd .. \
+    && rm mono -r
```
Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: can we do this `rm -rf mono`

Comment (vlesierse) [inline 1.0.0-beta4/Dockerfile]: This will reduces the image size by throwing away the source files

File: 1.0.0-beta4/Dockerfile (hunk 1 above)
Comment (vlesierse) [inline 1.0.0-beta4/Dockerfile]: I will remove this block

Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: @vlesierse I was just referring to the comment. env vars are still necessary, right?

File: 1.0.0-beta4/Dockerfile (hunk 2 above)
Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: @vlesierse yeah, but can you please reorder the rm command to look like `rm -rf xxx`

Comment (vlesierse) [inline 1.0.0-beta4/Dockerfile]: Ah, I see. Off course

File: 1.0.0-beta4/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+FROM mono:3.12
```
Comment (ahmetalpbalkan) [inline 1.0.0-beta4/Dockerfile]: @vlesierse do you think we need to use mono as base in this case where we're compiling/installing mono from source? in any case we should probably upgrade to `mono:4` here.

Comment (vlesierse) [inline 1.0.0-beta4/Dockerfile]: I tried it with debian:wheezy but it gave me a compile error. Probably I'm missing something. Let me try again and otherwise mono:4 makes more sense.

Comment (vlesierse) [inline 1.0.0-beta4/Dockerfile]: Mono doesn't compile without a mono base image. Actually kind a strange. I have changed the base image to `mono:4`

**Comment by vlesierse** [comment]
Done

**Comment by ahmetalpbalkan** [comment]
LGTM. Thanks @vlesierse. closes #50.
